falcon/testing/client.py

```diff
@@ -347,6 +347,13 @@ def json(self) -> Optional[Union[dict, list, str, int, float, bool]]:
 
         return json_module.loads(self.text)
 
+    def __repr__(self):
+        body_text = '{}'.format(self.text[:20])
+
+        return '<Status: {}, Content-Type: {}, Text: {}>'.format(
+            self.status_code, self.headers['Content-Type'], body_text
```

Comment:
What happens if the response payload is invalid UTF-8?
```python
>>> testing.Result([b'\xc3\x28'], falcon.HTTP_418, {})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/RioAtHome/falcon/falcon/testing/client.py", line 351, in __repr__
    body_text = '{}'.format(self.text[:20])
  File "/tmp/RioAtHome/falcon/falcon/testing/client.py", line 339, in text
    self._text = self.content.decode(encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 0: invalid continuation byte
```

It is IMHO better to prevent `__repr__` from raising almost at any cost, since it can be very confusing to debug (IIRC the popular `pytest` has some safeguard to handle even this scenario, but it ain't pretty).

Reply:
How about making `__repr__` log the content of the payload instead of text?